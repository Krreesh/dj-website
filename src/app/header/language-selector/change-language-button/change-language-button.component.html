<button class="dropdown-item changeLanguage"
        data-change-language-button
        (click)="useLanguage(language)">{{language | uppercase}}</button>
