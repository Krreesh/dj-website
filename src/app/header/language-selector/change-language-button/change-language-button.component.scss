@import "../../../../colors";

.dropdown-item {
  padding: .25rem 1.25rem;
}
