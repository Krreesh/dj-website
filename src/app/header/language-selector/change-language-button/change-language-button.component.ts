import {Component, Input} from '@angular/core';
import {LanguageService} from '../../../translation/language.service';

@Component({
  selector: 'app-change-language-button',
  styleUrls: ['./change-language-button.component.scss'],
  templateUrl: './change-language-button.component.html'
})
export class ChangeLanguageButtonComponent {

  @Input()
  language: string;

  constructor(private languageService: LanguageService) {
  }

  useLanguage(language: string): void {
    this.languageService.updateCurrentLanguage(language);
  }
}
