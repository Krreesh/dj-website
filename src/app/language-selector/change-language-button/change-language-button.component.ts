import {Component, Input} from '@angular/core';
import {LanguageService} from '../../translation/language-service';

@Component({
  selector: 'app-change-language-button',
  template: `
    <button class="dropdown-item" ngbDropdownItem (click)="useLanguage(language)">
      {{language}}
    </button>
  `,
  styles: []
})
export class ChangeLanguageButtonComponent {

  @Input()
  language: string;

  constructor(private languageService: LanguageService) {
  }

  useLanguage(language: string): void {
    this.languageService.setCurrentLanguage(language);
  }
}
