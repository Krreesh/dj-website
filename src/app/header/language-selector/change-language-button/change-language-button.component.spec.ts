import {ComponentFixture, TestBed} from '@angular/core/testing';
import {ChangeLanguageButtonComponent} from './change-language-button.component';
import {LanguageService} from '../../../translation/language.service';

describe('ChangeLanguageButtonComponent', () => {
  let component: ChangeLanguageButtonComponent;
  let fixture: ComponentFixture<ChangeLanguageButtonComponent>;
  let updateCurrentLanguage: jest.Mock;

  beforeEach(() => {
    updateCurrentLanguage = jest.fn();
    TestBed.configureTestingModule({
      declarations: [ChangeLanguageButtonComponent],
      providers: [
        {
          provide: LanguageService,
          useValue: {
            updateCurrentLanguage: updateCurrentLanguage
          }
        }
      ]
    });
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ChangeLanguageButtonComponent);
    component = fixture.debugElement.componentInstance;
  });

  it('should update the current language', () => {
    fixture.componentInstance.language = 'fr';
    fixture.detectChanges();

    findChangeLanguageButton().click();
    fixture.detectChanges();

    expect(updateCurrentLanguage).toHaveBeenCalledWith('fr');
  });

  function findChangeLanguageButton(): HTMLButtonElement {
    return fixture.nativeElement.querySelector('[data-change-language-button]');
  }
});
