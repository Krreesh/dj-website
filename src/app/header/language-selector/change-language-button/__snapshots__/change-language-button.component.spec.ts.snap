// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeLanguageButtonComponent should render the button 1`] = `
<app-change-language-button
  language={[Function String]}
  languageService={[Function Object]}
>
  <button
    class="dropdown-item changeLanguage"
    data-change-language-button=""
  >
    EN
  </button>
</app-change-language-button>
`;
